### PR TITLE
GRPC maintanance

### PR DESCRIPTION
**Pull request by olegbespalov**
# What?

Adjust dependencies to make it compatible with the latest k6.

# Why?

Currently, this extension doesn't compile with the latest k6.

### CODE REVIEW

**Review by mstoykov — APPROVED** [review]

**Comment by javaducky** [comment]
@olegbespalov, I am curious if there is any reason this hasn't merged yet. 

Some internal folks were having some build issues when I stumbled onto this PR (thank you!!!). I had them build from your branch, but didn't want to merge this change without your input.

**Comment by olegbespalov** [comment]
Hey @javaducky !

My bad :cry:  I should have explicitly mentioned the reason. I haven't merged it since the change still touches the logic, but there are no CI tests, and I don't know how to run some regression on that.

So I've postponed the merge decision for someone from Loki team who can quickly validate if the changes are legit and merge it. 

@chaudum I see your activity in this repo :sweat_smile:  Could you please maybe help with reviewing/testing the changes?

**Comment by javaducky** [comment]
> My bad 😢 I should have explicitly mentioned the reason. I haven't merged it since the change still touches the logic, but there are no CI tests, and I don't know how to run some regression on that.

All good @olegbespalov ! Thank you so much for making the fixes to this.  🙇

File: go.mod [hunk 1]
```diff
@@ -8,152 +8,48 @@ require (
...
-	google.golang.org/genproto v0.0.0-20230803162519-f966b187b2e5 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20230822172742-b8732ec3820d // indirect
 	google.golang.org/grpc v1.57.0 // indirect
 	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/guregu/null.v3 v3.5.0 // indirect
-	gopkg.in/yaml.v2 v2.4.0 // indirect
-	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
```
Comment (chaudum) [inline go.mod:155]: In Loki, we still pin grpc to v1.53.0 - but that might not be necessary any more.

**Review by chaudum — APPROVED** [review]
LGTM

**Pull request merged**